### DISCUSSION TITLE
How do I handle routing for Steps component

### DISCUSSION THREAD

**Discussion by scott-mac** [primefaces/community#840] [PrimeNG] [answered]
Sorry if this is a dumb question, I'm new to a lot of this. I am using the [Sakai ](https://github.com/primefaces/sakai-ng)dashboard template as my starting point for my project. I want to add a steps component one of my pages but I can't figure out how to do the routing so that each step loads its component correctly. The documentation shows this usage in the example (this is in the primary component where it declares the model for the steps) :

```
this.items = [{
                label: 'Personal',
                routerLink: 'personal'
            },
            {
                label: 'Seat',
                routerLink: 'seat'
            },
            {
                label: 'Payment',
                routerLink: 'payment'
            },
            {
                label: 'Confirmation',
                routerLink: 'confirmation'
            }
        ]; 
```
and then within the same folder the components ts/html for each step (personal, seat, payment, confirmation). 

I have set mine up exactly like this. The problem is with the routing. The Sakai template defines a routing module at every subfolder within the src, so for example, the demo has all the components under 'src/app/demo/components/uikit/' so you get a folder structure like this:

- uikit
  - uikit-routing.module.ts
  - uikit.module.ts
  - button
    - buttondemo-routing.module.ts
    - buttondemo.component.html
    - buttondemo.component.ts
    - buttondemo.module.ts
  - charts
    - chartsdemo-routing.module.ts
    - chartsdemo.component.html
    - chartsdemo.component.ts
    - chartsdemo.module.ts

At the root the uikit-routing.module.ts looks like this 

```
@NgModule({
    imports: [RouterModule.forChild([
        { path: 'button', data: { breadcrumb: 'Button' }, loadChildren: () => import('./button/buttondemo.module').then(m => m.ButtonDemoModule) },
        { path: 'charts', data: { breadcrumb: 'Charts' }, loadChildren: () => import('./charts/chartsdemo.module').then(m => m.ChartsDemoModule) },        
        { path: '**', redirectTo: '/notfound' }
    ])],
      exports: [RouterModule]
})
```

And within each child folder the routing module will look like this

```
@NgModule({
	imports: [RouterModule.forChild([
		{ path: '', component: ButtonDemoComponent }
	])],
	exports: [RouterModule]
})
export class ButtonDemoRoutingModule { }

```

Hopefully that makes sense, I'm at a loss over what to change or tweak to get this working, much obliged for any help you can provide, thanks.

**Comment by scott-mac** [comment, marked as answer]
figured it out, I got rid of the routing module at the level of the step component and then just used 
```

[RouterModule.forChild([
        { path: 'root', component: RootComponent, 
            children:[
                { path: 'step1', component: Step1Component },
                { path: 'step2', component: Step2Component },
            ] 
        }
    ])],
```

**Reply by cetincakiroglu** [reply]
Hi,

I'm happy to hear that, you can also use new angular.dev documentation which is great for starting angular :)
